packages/utils/src/baggage.ts

```diff
@@ -54,6 +56,23 @@ export function getThirdPartyBaggage(baggage: Baggage): string {
   return baggage[1];
 }
 
+/**
+ * Checks if baggage is mutable
+ * @param baggage
+ * @returns true if baggage is mutable, else false
+ */
+export function isBaggageMutable(baggage: Baggage): boolean {
+  return baggage[2];
+}
+
+/**
+ * Sets the passed baggage immutable
+ * @param baggage
+ */
+export function setBaggageImmutable(baggage: Baggage): void {
+  baggage[2] = false;
+}
```

Comment:
(FP nerdiness on) Did we evaluate here, if it's possible/feasable/sensical to return a new Baggage object/array here, instead of mutating the one we pass in? (FP nerdiness off)

Reply:
Haha I agree that it would be much nicer FP-wise. I think that this applies to many of the functions in the baggage API. So if we change it, I think we should change it for all functions that mutate baggage. Which overall makes me think that this should be its own PR because we'll probably have to make quite a few changes then. 

The one point I can see that makes this a little less desireable is that it increases bundles size (albeit not a lot). Not saying that it's not worth that. Just putting it out there.

@AbhiPrasad do you have thoughts on this?

Reply:
I typically favour the FP approach - but I would rather stick with mutation

1) less bytes
2) we have great test coverage
3) don't have to go back and change stuff now

If this ends up causing bugs and confusion, then we come back and revisit.